Sources/GRPCHTTP2Core/Client/HTTP2ClientTransport.swift

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCore
+
+/// A namespace for the HTTP/2 client transport.
+public enum HTTP2ClientTransport {}
+
+extension HTTP2ClientTransport {
+  /// A namespace for HTTP/2 client transport configuration.
+  public enum Config {}
+}
+
+extension HTTP2ClientTransport.Config {
+  public struct Compression: Sendable {
+    /// The default algorithm used for compressing outbound messages.
+    ///
+    /// This can be overridden on a per-call basis via ``CallOptions``.
+    public var algorithm: CompressionAlgorithm
+
+    /// Compression algorithms enabled for inbound messages.
+    ///
+    /// Note that ``CompressionAlgorithm/none`` is always supported, even if it isn't set here.
+    public var enabledAlgorithms: CompressionAlgorithmSet
+
+    /// Creates a new compression configuration.
+    ///
+    /// See also ``defaults``.
```

Comment:
Similar to the above comment, we can use `SeeAlso` here:
```suggestion
    /// - SeeAlso: ``defaults``.
```